### PR TITLE
tc_optops but with no lambda

### PR DESCRIPTION
**Pull request by chaosagent**
#3210 but with no lambda. did not use brain when writing this one.

also symbolic doesn't seem to automatically simplify `x // k * k + x % k` to `x`

### CODE REVIEW

**Comment by geohot** [comment]
Looks good, but there's a merge conflict on this.

File: tinygrad/codegen/linearizer.py [hunk 1]
```diff
@@ -259,11 +259,6 @@ def calc_tc_idxs(local_size: int, aliases: List[List[int]]):
                 full_var_sz *= next_var.max+1
             replace_idxs.append(full_var)
           return replace_idxs
-        replace_acc_idxs = calc_tc_idxs(tc.thread_local_sizes[2], tc.thread_local_aliases[2])
```
Comment (geohot) [inline tinygrad/codegen/linearizer.py:262]: Why is this removed?

**Comment by geohot** [comment]
I'm also starting to think we don't have enough testing around tensor cores to merge anything related to them.

There's a whole bunch of subtle ways this can be wrong? Did you test it on Mac and AMD?

File: tinygrad/codegen/linearizer.py (hunk 1 above)
Comment (chaosagent) [reply tinygrad/codegen/linearizer.py:262]: before, afaict the only thing that this did was permute the output axes for hip tc to work around upcasts always coming before locals (it was building a 16-length index on a 8-length shape dimension, even). now we specify the output pattern through optops.

maybe i missed something about why this was here before.

**Comment by chaosagent** [comment]
I tested on AMD on simple_matmul and cifar. I don't have a mac to test on, unfortunately. I think we can test AMD tc with remu in CI.

**Comment by chaosagent** [comment]
actually, this one is still relevant and doesn't require any additional tests

**Comment by geohot** [comment]
Closing, stale.
